### ISSUE TITLE
[NEXT-354] [appDir]: Prod deployment breaks client side navigation, but works with local prod build

### ISSUE THREAD

**Issue by suneettipirneni** [vercel/next.js#44728]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
  Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:08:47 PST 2022; root:xnu-8792.61.2~4/RELEASE_X86_64
    Binaries:
      Node: 16.16.0
      npm: 8.11.0
      Yarn: 3.2.4
      pnpm: N/A
    Relevant packages:
      next: 13.1.2-canary.3
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue

https://github.com/discordjs/discord.js/tree/main/apps/website

### To Reproduce

You can visit this link to see the issue in action: https://discordjs.dev/docs/packages/core/main

Or you can clone the code and deploy it manually on vercel:

`git clone https://github.com/discordjs/discord.js.git`

Deploy on vercel, set the following settings:
- build command: `yarn build:prod`
- root directory: `apps/website`, include outside source files

Navigate to the following path: `/docs/packages/core/main`

Click on any of the sidebar items.

### Describe the Bug

When clicking any of the link items in the sidebar the page does a full reload. However this behavior does not occur in a prod build running locally.

The discrepancy is shown below between both of the deployments. 

Vercel deployment:

https://user-images.githubusercontent.com/77477100/211343621-af5d14a3-3d66-4942-8cc6-aa418bf18009.mov

Local prod build `next build` and `next start`:

https://user-images.githubusercontent.com/77477100/211343830-8860c194-e800-453f-a5c8-63b1ac12f7cc.mov

### Expected Behavior

Both builds and deployments should use client-side routing, but the vercel deployment does not.

### Which browser are you using? (if relevant)

Chrome 108.0.5359.124 (Official Build) (x86_64)

### How are you deploying your application? (if relevant)

Vercel

<sub>[NEXT-354](https://linear.app/vercel/issue/NEXT-354/[appdir]-prod-deployment-breaks-client-side-navigation-but-works-with)</sub>

**Comment by james-elicx** [comment]
I just want to note that I have been experiencing this issue as well when using the edge runtime and the link element is in a client component. It appears to be fine for me when using the nodejs runtime or the link element is in a server component, but if I change to the edge runtime and a client component, clicking links does a full page reload.

**Comment by trulymittal** [comment]
Seems to be a bug, works in the local build but not on Vercel deployment...

So here is the issue...The **Dynamic routes seem to be a problem**, the path which have been generated using `generateStaticParams` worked with the `Link`, but the paths that was not generated at build time reloads the page...

Here is the build output from Vercel, where output shows what paths are generated at build time:

```bash
info  - Collecting page data...
--
17:40:26.842 | [
17:40:26.842 | { slug: 'another-one' },
17:40:26.842 | { slug: 'usecallback-hook-in-react' },
17:40:26.842 | { slug: 'what-is-rest-api' },
17:40:26.842 | { slug: 'my-first-blog' }
17:40:26.842 | ]
17:40:26.863 | info  - Generating static pages (0/7)
```

The deployment is on this URL: https://sanity-studio-v3-nextjs.vercel.app/
here you can validate that only the paths that are generated at build time **does not reload the page** but other **dynamic routes does reload the page**...

**Comment by trulymittal** [comment]
@max-ch9i Its, not a workaround, its an insight to one of the current problem with `Link`

**Comment by ijjk** [comment]
Hi, a fix for this has been opened, once it has been released we will update here and you can re-deploy your project without any other changes necessary.

**Comment by suneettipirneni** [comment]
Hi @ijjk, thanks a bunch for the fix. I went to your PR on vercel and tested the preview deployment and I noticed that while the client-side routing seems to be fixed, parent layouts (in this case the sidebar) seems to be completely re-rendered rather than persisted when clicking on a in the sidebar link, as seen by the scroll position being reset. This behavior doesn't occur on a local prod build, should I open a separate ticket for this issue?
